### ISSUE TITLE
How to intergate with self model，such as tencent hunyuan, ailibaba qianwen。

### ISSUE THREAD

**Issue by Rainmen-xia** [microsoft/genaiscript#855]
How to intergate with self model，such as tencent hunyuan, qianwen。

**Comment by Rainmen-xia** [comment]
yes。

**Comment by pelikhan** [comment]
@Rainmen-xia https://github.com/microsoft/genaiscript/pull/888

**Comment by pelikhan** [comment]
Alibaba was added.

**Comment by anrgct** [comment]
You can call it using OpenAI-compatible endpoints. If you want to be compatible with function calls, you need to [modify the code](https://www.cnblogs.com/mesopotamiaa/p/18555652) or use [fallback tools](https://microsoft.github.io/genaiscript/blog/fallback-tools).

**Comment by pelikhan** [comment]
@anrgct  are these mods for alibaba? I could enable fallback-tools automatically for those?

**Comment by anrgct** [comment]
Are these mods for Alibaba? Well, that's just my hack code because it's not fully compatible with the OpenAI format's function call modifications.
I could enable fallback-tools automatically for those? Maybe yes? I found that most models' function call capabilities are quite poor, but using fallback-tools works very well, like qwen2.5-coder-32b.

**Comment by pelikhan** [comment]
Yes we can definitely enable fallback on those.

**Comment by yaonan** [comment]
The usage of Alibaba Cloud API KEY in the `getting-started/configuration` document is inaccurate. 

The value of the environment variable `ALIBABA_API_KEY` should use the `Bearer sk-...` format instead of `sk_...`, otherwise, the following error will occur

```
genaiscript: error
LLM error (401): You didn't provide an API key. You need to provide your API key in an Authorization header using Bearer auth (i.e. Authorization: Bearer YOUR_KEY).
```

**Comment by pelikhan** [comment]
Thanks for pointing this out. Fix coming.

**Comment by pelikhan** [comment]
fixed in 1.86.2
